src/github.com/matrix-org/dendrite/clientapi/auth/storage/accounts/filter_table.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2017 Jan Christian Grünhage
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package accounts
+
+import (
+	"database/sql"
+	"context"
+	"fmt"
+)
+
+const filterSchema = `
+-- Stores data about filters
+CREATE TABLE IF NOT EXISTS account_filter (
+	-- The filter
+	filter TEXT NOT NULL,
+	-- The ID
+	id SERIAL UNIQUE,
```

Comment:
SERIAL means that postgres will just increment that forever, which leaks a tiny bit of data to users creating filters, namely how many filters have been created on the server. Synapse does something similar, but per user.

Reply:
Seems reasonable. (If we wanted to be clever here we could use a SHA-256 hash of the filter as the filter ID. That way we wouldn't get duplicates if the clients uploaded the same filter multiple times.)

Reply:
We would still get duplicates for multiple users though.. That this is at all user specific is somewhat displeasing, it would be nicer if there was just an ID and the associated filter, completely leaving out the users. @NegativeMjark Do you think I should use SHA256 as the ID or keep it as it is currently?

Reply:
I think ID is fine for now